src/poetry/packages/locker.py

```diff
@@ -129,8 +129,23 @@ def locked_repository(self) -> Repository:
                 hashes = cast("dict[str, Any]", metadata["hashes"])
                 package.files = [{"name": h, "hash": h} for h in hashes[name]]
             else:
-                files = metadata["files"][name]
-                package.files = files
+                if source_type in {"git", "directory", "url"}:
+                    package.files = []
+                else:
+                    files = metadata["files"][name]
+                    if source_type == "file":
+                        if len(files) > 1:
+                            filename = Path(url).name
+                            for file_item in files:
```

Comment:
suggest
```python
filename = Path(url).name
package.files = [item for item in files if item["file"] == filename]
```
in the expectation that there is only one match, but if multiple then we don't know which is right so had better take them all